interface/wx/webrequest.h

```diff
@@ -0,0 +1,639 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        webrequest.h
+// Created:     2018-10-14
+// Copyright:   (c) 2018 wxWidgets development team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+/**
+    @class wxWebRequest
+
+    This class allows for simple HTTP requests using the operating system's
+    components as implementation.
+
+    The latest features of the operating system will be used if available
+    (e.g. HTTP/2, TLS 1.3).
+    System-wide configuration like proxy and SSL certificates will be used
+    when possible.
+
+    Instances of wxWebRequest are created by using
+    wxWebSession::CreateRequest().
+
+    The requests are handled asynchronously and event handlers are used to
+    communicate the request status. The response data may be stored in
+    memory, to a file or processed directly, see SetStorage() for details.
+
+    Example usage:
+
+    @code
+        // Create the request object
+        wxObjectDataPtr<wxWebRequest> request(
+            wxWebSession::GetDefault().CreateRequest("https://www.wxwidgets.org/downloads/logos/blocks.png"));
+
+        // Bind state event
+        request->Bind(wxEVT_WEBREQUEST_STATE, [](wxWebRequestEvent& evt) {
+            switch (evt.GetState())
+            {
+                // Request completed
+                case wxWebRequest::State_Completed:
+                {
+                    wxImage logoImage(*evt->GetResponse()->GetStream());
+                    if (logoImage.IsOK())
+                        wxLogInfo("Image loaded");
+                    break;
+                }
+                // Request failed
+                case wxWebRequest::State_Failed:
+                    wxLogError("Could not load logo: %s", evt.GetErrorDescription());
+                    break;
+            }
+
+        });
+
+        // Start the request
+        request->Start();
+    @endcode
+
+    @section descriptions Implementation Descriptions
+
+    The following APIs are used per platform, additional details
+    about supported features may be found in their documentation.
+
+    Available features by implementation and minimum version:
+    <table>
+    <tr><th>Operating System</th><th>API</th><th>HTTPS</th><th>HTTP/2</th></tr>
+    <tr>
+        <td>Windows</td>
+        <td>
+            <a target=_new href="https://docs.microsoft.com/en-us/windows/desktop/WinHttp/about-winhttp">WinHTTP</a>
+        </td>
+        <td>Yes</td>
+        <td>Windows 10 1607</td>
```

Comment:
This isn't right, is it? Some features are only available since this version, but most things work even in much older versions, don't they?